aws-sdk-resources/lib/aws-sdk-resources/services/s3/file_downloader.rb

```diff
@@ -135,31 +133,44 @@ def map_files(slice, dir, mode)
         end
       end
 
-      def get_range(chunks)
+      def clean_up_parts(parts)
+        parts.each do |filename|
+          File.unlink(filename) if File.exists?(filename)
+        end
+      end
+
+      def multithreaded_get_by_ranges(chunks)
         thread_batches(chunks, 'range')
       end
 
-      def get_part(parts)
+      def multithreaded_get_by_parts(parts)
         thread_batches(parts, 'part_number')
       end
 
       def thread_batches(chunks, param)
         batches = file_batches(chunks, param)
+        parts = batches.inject([]) {|a, batch| a.push(*batch.values); a}
```

Comment:
`Array#push` returns the modified receiver, so this could be just

```rb
parts = batches.inject([]) {|a, batch| a.push(*batch.values) }
```

Even better, you could use `Array#flat_map`:

```rb
parts = batch.flat_map(&:values)
```